lib/ddtrace/buffer.rb

```diff
@@ -73,6 +51,75 @@ def close
       end
     end
 
+    protected
+
+    def full?
+      @max_size > 0 && @items.length >= @max_size
```

Comment:
don't you need the `@mutex` here?

Reply:
Shouldn't need to; these "protected" methods are not public, and are wrapped by public methods that use `@mutex`.